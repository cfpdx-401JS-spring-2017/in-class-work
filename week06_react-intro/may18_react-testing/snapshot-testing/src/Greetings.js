import React, { Component } from 'react';

export function Greeter ({ name }) {
  return <div>Hello <span>{name}</span></div>;
}

export class Greetings extends Component {
  render() {
    const { people } = this.props;
    return (
      <ul>
        {people.map((person, i) => <Greeter key={i} name={person}/>)}
      </ul>
    );
  }
}