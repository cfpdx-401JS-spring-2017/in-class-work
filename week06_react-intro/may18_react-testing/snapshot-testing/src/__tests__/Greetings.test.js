import React from 'react';
import renderer from 'react-test-renderer';
import { Greeter, Greetings } from '../Greetings';

describe('Greeter', () => {

    it('says hello to Nicky', () => {
        const component = renderer.create(<Greeter name="Nicky" />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
})

describe('Greetings', () => {

    it('says hello to people', () => {
        const component = renderer.create(
            <Greetings people={['Jill', 'Julie', 'Jonsie']} />
        );
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
})
