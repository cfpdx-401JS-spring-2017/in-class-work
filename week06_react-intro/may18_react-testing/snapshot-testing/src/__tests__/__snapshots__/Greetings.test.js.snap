exports[`Greeter says hello to Nicky 1`] = `
<div>
  Hello 
  Nicky
</div>
`;

exports[`Greetings says hello to people 1`] = `
<ul>
  <div>
    Hello 
    Jill
  </div>
  <div>
    Hello 
    Julie
  </div>
  <div>
    Hello 
    Jonsie
  </div>
</ul>
`;
